### PR TITLE
Bug fix: Validation of fields in signup form

### PR DESCRIPTION
**Pull request by dmnisson**
https://app.asana.com/0/775116702697292/1118106252500976/f

This PR adds client-side validation for immediate feedback if user enters invalid values into signup fields. This feedback is implemented using the following changes:

* Created a system that allows multiple error messages to be displayed instead of just one
* Used v-bind:class to highlight fields with invalid input
* Moved submission logic from submit button's on-click handler, which bypasses browser validation, to the form's on-submit handler.
* Put validation logic in the on-click handler, and prevent form submission only if there are errors. In the future, this will allow the browser to check for errors if someone adds a new field and forgets to add validation logic to the on-click handler.
* Fields that are required on signup are now sent to auth/register instead of api/user, so that they can be verified before the user is saved in the database.

### CODE REVIEW

**Comment by jkeat** [comment]
FYI: @dmnisson discovered that even though we require & validate the signup fields on the frontend, we don't do that on the backend.

David is going to investigate this a bit more. If it looks fairly high-effort, we'll track it in a separate bug. If it'll be a quick fix, he'll make a companion PR in the server repo to address this issue.

**Comment by dmnisson** [comment]
The Asana link was incorrect. I've edited it to point to the correct task.

**Comment by georgiagallant333** [comment]
• Some signup errors were caught on the server side and some on the frontend. Depending on where they surfaced they were formatted differently– the updated errors that were just implemented were caught on the front end and displayed in red above the email box, but errors that already existed like a password < 8 characters, and an email already taken were caught on the server side and displayed in black below form boxes. Made those catch on the front end first and made all errors uniform so they are red about the email box. 
• Made all password errors appear at the same time and bulleted under text saying: "password must have at least:"
• Simplified checking for caps, lowercase, and numbers

File: src/components/Registration/VolunteerSignupForm.vue [hunk 1]
```diff
@@ -113,6 +131,7 @@
           <td colspan="2">
             <input
               class="form-control"
+              v-bind:class="{'form-control-invalid': invalidInputs.indexOf('favoriteAcademicSubject') > -1}"
```
Comment (jkeat) [inline src/components/Registration/VolunteerSignupForm.vue]: Later on, we should make something like a `form-field` component to cut down on the repetition of logic. But for now, looks good :)

**Pull request merged**